templates/common/infra/bicep/core/host/container-app.bicep

```diff
@@ -5,24 +5,46 @@ param tags object = {}
 param containerAppsEnvironmentName string
 param containerName string = 'main'
 param containerRegistryName string
+param secrets array = []
 param env array = []
 param external bool = true
 param imageName string
-param keyVaultName string = ''
-param managedIdentity bool = !empty(keyVaultName)
 param targetPort int = 80
 
+@description('User assigned identity name')
+param identityName string = ''
+
 @description('CPU cores allocated to a single container instance, e.g. 0.5')
 param containerCpuCoreCount string = '0.5'
 
 @description('Memory allocated to a single container instance, e.g. 1Gi')
 param containerMemory string = '1.0Gi'
 
+resource userIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2023-01-31' existing = {
+  name: identityName
+}
+
+module containerRegistryAccess '../security/registry-access.bicep' = {
+  name: '${deployment().name}-registry-access'
+  params: {
+    containerRegistryName: containerRegistryName
+    principalId: userIdentity.properties.principalId
+  }
+}
+
```

Comment:
registry-access guid needs to have more variants, ran into conflict issues upon redeploy.

I'm switching to using roles.bicep which has

  name: guid(subscription().id, resourceGroup().id, principalId, roleDefinitionId)